storage/interface.go

```diff
@@ -161,12 +161,12 @@ type LabelQuerier interface {
 	// It is not safe to use the strings beyond the lifetime of the querier.
 	// If matchers are specified the returned result set is reduced
 	// to label values of metrics matching the matchers.
-	LabelValues(ctx context.Context, name string, matchers ...*labels.Matcher) ([]string, annotations.Annotations, error)
+	LabelValues(ctx context.Context, name string, hints *LabelHints, matchers ...*labels.Matcher) ([]string, annotations.Annotations, error)
```

Comment:
Changing this interface seems like quite a big step; it forces a change on all downstream implementations.

Reply:
Thanks for reviewing. I was also hesitant to change the interface initially. But only other way I could think of passing the limit to storage was through the `context`.

I also noticed that the interface was changed a few times in the past to add the matchers to the API.
- eg: https://github.com/prometheus/prometheus/pull/8400

Is this a blocker? Does it need more discussions with the maintainers?